machine-learning/solutions/seismic_salt_detection/trainer/model.py

```diff
@@ -73,51 +73,40 @@ def create_regressor(config, parameters):
     Returns:
         A configured and ready to use `tf.estimator.DNNRegressor`
     """
-
     # Mean and Standard Deviation Constants for normalization.
-    with file_io.FileIO(parameters.mean_path, mode='r') as f:
-        mean = pickle.load(f)
-    with file_io.FileIO(parameters.std_path, mode='r') as f:
-        std = pickle.load(f)
+    mean = np.float32(parameters.depth_mean)
+    std = np.float32(parameters.depth_std)
 
     # Columns to be used as features.
-    hour = tf.feature_column.categorical_column_with_identity(
-        'hour', num_buckets=24)
-    hour = tf.feature_column.embedding_column(
-        hour, dimension=parameters.hour_embedding)
-
-    day = tf.feature_column.categorical_column_with_identity(
-        'day', num_buckets=7)
-    day = tf.feature_column.embedding_column(
-        day, dimension=parameters.day_embedding)
-
-    weather = [tf.feature_column.numeric_column(
-        'weather' + str(i),
-        normalizer_fn=(lambda x, i = i: (x - mean[i]) / std[i])
-    ) for i in range(constants.WEATHER_SIZE)]
-
-    distribution = [tf.feature_column.numeric_column(
-        'distribution' + str(i)
-    ) for i in range(constants.DISTRIBUTION_SIZE)]
-
-    feature_cols = [hour, day] + weather + distribution
+    
+    depth = tf.feature_column.numeric_column(
+        'depth',
+        normalizer_fn=(lambda x: (x - mean) / std))
+    
+    image = hub.image_embedding_column('image', parameters.module)
+    
+    feature_cols = [depth,image]
 
-    # Evaluation metric.
-    def mean_absolute_error(labels, predictions):
-        """Creates mean absolute error metric.
+    def estimator_metrics(labels, predictions):
+        """Creates metrics for Estimator.
 
-        Metric is used to evaluate the model.
+        Metrics are used to evaluate the model.
 
         Args:
             labels: Evaluation true labels.
             predictions: Evaluation model predictions.
 
         Returns:
-            A dictionary with the evaluation metric
+            A dictionary with the evaluation metrics
         """
-        pred_values = predictions['predictions']
-        return {'mae': tf.metrics.mean_absolute_error(
-            labels, pred_values)}
+        pred_logistic = predictions['logistic']
```

Comment:
This might not be the right place to comment, but it's non-obvious to me what 'logistic' by its name is supposed to represent.

Reply:
'logistic' is what TF API expects as an input to give you back probabilities for the classification predictions. This is documented in the TF API.

Reply:
Bleh.  That's an unfortunate name for (I'm assuming) "Tensor of weights to which a logistic function has been applied."  But it sounds like from what you're saying, it's not even that but is "Tensor of weights to which a softmax has been applied."  Is that about right?

Also, and continuing my own education, where is predictions generated?  Is 'logistic' a magic string?